diff --git a/node_modules/the-core-ui-module-tdslideshow/src/components/AddPhoto.js b/node_modules/the-core-ui-module-tdslideshow/src/components/AddPhoto.js
index f41ddff..e8454fa 100644
--- a/node_modules/the-core-ui-module-tdslideshow/src/components/AddPhoto.js
+++ b/node_modules/the-core-ui-module-tdslideshow/src/components/AddPhoto.js
@@ -28,7 +28,7 @@ const AddPhoto = ({
   // ******* SETUP *******
   const [cameraRef, setCameraRef] = useState();
   const [cameraType, setCameraType] = useState("back");
-  const { getHeight, fontSize } = ScaleHook();
+  const { getHeight, getWidth, fontSize } = ScaleHook();
 
   // ******* STYLES *******
   const styles = {
diff --git a/node_modules/the-core-ui-module-tdslideshow/src/components/ImageSlider.js b/node_modules/the-core-ui-module-tdslideshow/src/components/ImageSlider.js
index 1bbdec1..2e8d616 100644
--- a/node_modules/the-core-ui-module-tdslideshow/src/components/ImageSlider.js
+++ b/node_modules/the-core-ui-module-tdslideshow/src/components/ImageSlider.js
@@ -33,11 +33,12 @@ const ImageSlider = ({
   sliderSpacerHeight,
   sliderIcon,
   DateSelectors,
+  CustomButton
 }) => {
   // ******* SETUP *******
   const screenWidth = Dimensions.get("screen").width;
   const leftAlign = (screenWidth - imageWidth) / 2;
-  const { getHeight, getWidth } = ScaleHook();
+  const { getHeight, getWidth, fontSize } = ScaleHook();
 
   // ******* STYLES *******
   const styles = StyleSheet.create({
